conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+    def pkg_format_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = ["2", None]
+        if v not in valid_values:
+            raise ValueError(
+                f"Invalid value {v} for pkg_format. Valid values are {valid_values}"
+            )
+        return v
+
+
+class CondaForgeDocker(BaseModel):
+    model_config = ConfigDict(extra="allow")
+
+    executable: str = Field(
+        description="The executable for Docker", default="docker"
+    )
+
+    fallback_image: str = Field(
+        description="The fallback image for Docker",
+        default="quay.io/condaforge/linux-anvil-comp7",
+    )
+
+    command: str = Field(
+        description="The command to run in Docker", default="bash"
+    )
+
+    interactive: bool = Field(
+        description="Whether to run Docker in interactive mode", default=None
+    )
+
+    image: str = Field(description="The image for Docker", default=None)
+
+
+class ShellCheck(BaseModel):
+    enabled: bool = Field(
+        description="Whether to use shellcheck to lint shell scripts",
+        default=False,
+    )
+
+
+class ConfigModel(BaseModel):
+    appveyor: Dict[str, Any] = Field(
+        default={"image": "Visual Studio 2017"},
+        description="AppVeyor CI settings. This is usually read-only and should not normally be manually modified. Tools like conda-smithy may modify this, as needed.",
+    )
+
+    azure: AzureConfig = AzureConfig()
+
+    bot: BotConfig = BotConfig()
+
+    build_platform: Annotated[
+        Union[Dict[Platforms, Platforms], None],
+        Field(
+            description="This is a mapping from the target platform to the build platform for the package to be built."
+        ),
+    ] = {p.value: p.value for p in Platforms}
+
+    build_with_mambabuild: Annotated[
+        Union[bool, None],
+        Field(
+            description="configures the conda-forge CI to run a debug build using the mamba solver. More information can be in the [mamba docs](https://conda-forge.org/docs/maintainer/maintainer_faq.html#mfaq-mamba-local)."
+        ),
+    ] = True
+
+    channel_priority: Annotated[
+        Union[ChannelPriorityConfig, None],
+        Field(
+            description="The channel priority level for the conda solver during feedstock builds. This can be one of `strict`, `flexible`, or `disabled`. For more information, see the [Strict channel priority](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority) section on conda documentation."
+        ),
+    ] = None
+
+    channels: CondaForgeChannels = CondaForgeChannels()
+
+    choco: Annotated[
+        Union[List[str], None],
+        Field(
+            description="This parameter allows for conda-smithy to run chocoloatey installs on Windows when additional system packages are needed. This is a list of strings that represent package names and any additional parameters."
+        ),
+    ] = []
```

Comment:
Mutable defaults should use `Field` with `default_factory`. Use `Field` in value, not Annotated.